src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp

```diff
@@ -604,18 +605,29 @@ void AlgorithmPaintToolBox::import()
 
 void AlgorithmPaintToolBox::updateView()
 {
+    setButtonsDisabled(true);
     medAbstractView *view = this->getWorkspace()->stackedViewContainers()->getFirstSelectedContainerView();
 
     if (view)
     {
-        setCurrentView(qobject_cast<medAbstractImageView*>(view));
-
-        updateMouseInteraction();
-
-        medAbstractData* data = currentView->layerData(0);
-
-        if(data)
+        medAbstractImageView * v = qobject_cast<medAbstractImageView*>(view);
+        if (v)
         {
+            // AlgorithmPaintToolBox does not work with meshes
+            for (unsigned int i=0 ; i<v->layersCount() ; ++i)
+            {
+                medAbstractData *data = v->layerData(i);
+                if(!data || data->identifier().contains("vtkDataMesh"))
+                {
+                    qDebug()<<"AlgorithmPaintToolBox::updateView() do not use meshes in this toolbox.";
+                    return;
+                }
```

Comment:
Indeed it seems to be a good idea I will do that maybe later this afternoon or on Monday